src/clj_foundation/conversions.clj

```diff
@@ -0,0 +1,48 @@
+(ns clj-foundation.conversions
+  (:require [clj-foundation.errors :refer [failure? must-be]]
+            [clj-foundation.patterns :as patterns]))
+
+
+(def Map
+  "Alias for clojure.lang.IPersistentMap for use in type conversions"
+  clojure.lang.IPersistentMap)
+
+(def Vector
+  "Alias for clojure.lang.PersistentVector"
+  clojure.lang.PersistentVector)
+
+
+(defmulti convert
+  "Convert src-instance to dest-class if possible.  Returns patterns/NO-RESULT-ERROR
+  on failure.  For examplle:
+
+  * (convert Boolean/TYPE \"true\")
+  * (convert Map vararg-parameter-kvs)"
+  (fn [dest-class src-instance] [dest-class (class src-instance)]))
+
+
+(defmethod convert [java.io.File String] [_ str]
+   (java.io.File. str))
+
+
+(defmethod convert [Boolean/TYPE String] [_ str]
```

Comment:
Not sure how you're using this yet, but I would add support for int values 0, 1, and string values "0" and "1".